src/focus.service.ts

```diff
@@ -588,43 +598,49 @@ export class FocusService {
    * Returns if the element can receive focus.
    */
   private isFocusable(el: HTMLElement): boolean {
-    const record = this.registry.find(el);
-    if (record && record.excludeThis && record.excludeThis()) {
-      return false;
-    }
-
     const tabIndex = el.getAttribute('tabIndex');
 
     if (!!tabIndex && Number(tabIndex) < 0) {
       return false;
     }
 
-    const style = window.getComputedStyle(el);
-    if (style.display === 'none' || style.visibility === 'hidden') {
-      return false;
-    }
+    if (!this.isVisible(el)) { return false; }
+
+    if (this.focusTabIndexOnly) { return true; }
 
-    for (let parent = el; parent; parent = parent.parentElement) {
-      const parentRecord = this.registry.find(parent);
-      if (parentRecord && parentRecord.exclude && parentRecord.exclude()) {
+    // Eventually depricate arc-exclude as it is rarely used but consumes CPU cycles
```

Comment:
Can we benchmark this? I very much doubt this is a bottleneck, transversing up the tree should run at many millions of ops/sec